caddyhttp/markdown/markdown_test.go

```diff
@@ -79,27 +77,33 @@ func TestMarkdown(t *testing.T) {
 		}),
 	}
 
-	req, err := http.NewRequest("GET", "/blog/test.md", nil)
-	if err != nil {
-		t.Fatalf("Could not create HTTP request: %v", err)
-	}
-
-	rec := httptest.NewRecorder()
-
-	md.ServeHTTP(rec, req)
-	if rec.Code != http.StatusOK {
-		t.Fatalf("Wrong status, expected: %d and got %d", http.StatusOK, rec.Code)
+	get := func(url string) string {
+		req, err := http.NewRequest("GET", url, nil)
+		if err != nil {
+			t.Fatalf("Could not create HTTP request: %v", err)
+			return ""
+		}
+		rec := httptest.NewRecorder()
+		code, err := md.ServeHTTP(rec, req)
+		if err != nil {
+			t.Fatal(err)
+			return ""
+		}
+		if code != http.StatusOK {
+			t.Fatalf("Wrong status, expected: %d and got %d", http.StatusOK, code)
+			return ""
```

Comment:
Oh, really? So if you take out these returns, gometalinter throws the ineffective assignment error?

That's probably a bug in ineffassign. (Is it reproducible?) I won't require it of you, and will merge this PR as-is, but you are welcome to open a bug [upstream in the ineffassign repo](https://github.com/gordonklaus/ineffassign).

So, let me know if this is ready to be merged then, and I will do it.